nucypher/network/server.py

```diff
@@ -421,3 +421,19 @@ def __init__(self,
         elif public_certificate_filepath:
             kwargs['keypair'] = HostingKeypair(certificate_filepath=public_certificate_filepath, host=host)
         super().__init__(*args, **kwargs)
+
+
+class NonTLSHost():
```

Comment:
I'm just gonna go ahead and reject on the basis that TLS should always be enabled on _any_ host or endpoint for us.

Reply:
"Why? There's no sensitive information going over the wire."

Without TLS, anyone can inject anything into the traffic, local or external network. This violates our threat model, imho.

Reply:
I agree with you and I don’t know the history of why it was done this way in the 1st place.  But this isn’t the “upgrade Moe and Felix to https” pull request. Should I do that?

Reply:
i always assumed that for these “chaotic” characters they could just have their https endpoints be a load balancer or something

Reply:
Hmmm, I see your point about the load balancer, but I don't think that's a good assumption. We shouldn't be building a pretty major foot-gun like that.

I would much prefer to approve a PR that used TLS than not.

Reply:
I think it should be optional because  self signed TLS may not be preferable to a ca signed load balancer TLS which might be easier with a normal http endpoint.

Reply:
Unless you're going to build support in and make it a workable default (ie: working with LetsEncrypt), then I'm not sure there's currently any secure preference at all because a user wouldn't see a `NonTLSHost` class as a secure default.

Reply:
My opinion about this is that unless it's a CA signed domain validated cert, users will see **THIS SITE IS INSECURE!!! GO BACK!**.  right?

Reply:
Consider that ethereum client nodes run an RPC server, which can be interfaced with the HTTP web3.py provders here: <https://web3py.readthedocs.io/en/stable/providers.html#httpprovider>.  

For example, `geth` by default runs it's RPC server over a (non TLS) HTTP transport.  What is the thinking behind this design? 

On driver of my understanding here is the imperfect match between HTTP and the "web3 universe".  The web3 model of security is different in some distinct ways, since any blockchain read operation requires a cryptographic signature and all the transport has to do is pass along RPC calls.

Reply:
Right.  That is our universe.